### PR TITLE
Documentation Update for Jigawatt

### PR DESCRIPTION
**Pull request by CMoncur**

### CODE REVIEW

**Comment by scull7** [comment]
Conflicts 😢

File: README.md [hunk 1]
```diff
@@ -1,95 +1,269 @@
...
-  - returns an object to become the new `req.data` value
+### Basic Usage Example
+
+For this example, we have a poll and poll responses stored in a database. Consider the following endpoint:
 
-### Example Middleware
 ```
-const _         = require('ramda')
-const Bluebird  = require('bluebird')
+app.get('/best-city-results/:id', (req, res) => {
```
Comment (scull7) [inline README.md]: The code example should look something like so:
```javascript

const R = require('ramada') // use `R` for Ramda

app.get('/best-city-results/:id', JW(pollDetails, formatPoll)
```

Jigawatt middleware are *not* thenable (Promise) returning functions; they are simple javascript objects that have one or more of the following methods:  `awesomize`, `io` and/or `transform`.  If you would like to use a Jigawatt middleware as a Promise there is the `JW.promisify` method.

File: README.md [hunk 2]
```diff
@@ -1,95 +1,269 @@
...
+    poll    : data.poll.title
+  , results : tallyVotes(
+                getCity(data.poll)
+              , getAnswer(data.responses)
+              )
+  })
 }
+```
+
+Our Jigawatt middleware returns a thenable, which can be used to handle the resulting data as desired:
```
Comment (scull7) [inline README.md]: This is not true.

File: README.md [hunk 3]
```diff
@@ -1,95 +1,269 @@
...
-                              // present to user
-))
 ```
+const isCorrectLength = (str) => R.equals(24, str.length)
+
+...
+
+awesomize: (v) => ({
+  pollId : {
+    sanitize : R.toLower
```
Comment (scull7) [inline README.md]: functions must be passed as an array to `sanitize`, `validate` or `normalize`

**Comment by scull7** [comment]
Go ahead and commit your example program into the repository under an `example/basic` folder.

File: README.md (hunk 2 above)
Comment (jxbadam) [inline README.md]: Just a plain object, to get the thenable you need to do JW.promisify... I think.

**Pull request merged**